appsTransport_test.go

```diff
@@ -111,33 +112,76 @@ func TestJWTExpiry(t *testing.T) {
 }
 
 func TestCustomSigner(t *testing.T) {
-	check := RoundTrip{
-		rt: func(req *http.Request) (*http.Response, error) {
-			h, ok := req.Header["Authorization"]
-			if !ok {
-				t.Error("Header Accept not set")
-			}
-			want := []string{"Bearer hunter2"}
-			if diff := cmp.Diff(want, h); diff != "" {
-				t.Errorf("HTTP Accept headers want->got: %s", diff)
-			}
-			return nil, nil
+	tc := []struct {
+		nm           string
+		option       AppsTransportOption
+		bearerSuffix string
+	}{
+		{
+			nm:           "context-free signer",
+			option:       WithSigner(&noopSigner{}),
+			bearerSuffix: "",
+		},
+		{
+			nm:           "context signer",
+			option:       WithContextSigner(&noopSigner{}),
+			bearerSuffix: ":context",
 		},
 	}
 
-	tr, err := NewAppsTransportWithOptions(check, appID, WithSigner(&noopSigner{}))
-	if err != nil {
-		t.Fatalf("NewAppsTransportWithOptions: %v", err)
-	}
+	for _, c := range tc {
+		t.Run(c.nm, func(t *testing.T) {
+			check := AuthCaptureRoundTripper{}
 
-	req := httptest.NewRequest(http.MethodGet, "http://example.com", new(bytes.Buffer))
-	if _, err := tr.RoundTrip(req); err != nil {
-		t.Fatalf("error calling RoundTrip: %v", err)
+			tr, err := NewAppsTransportWithOptions(&check, appID, c.option)
+			if err != nil {
+				t.Fatalf("NewAppsTransportWithOptions: %v", err)
+			}
+
+			req := httptest.NewRequest(http.MethodGet, "http://example.com", new(bytes.Buffer))
+			req = req.WithContext(
+				context.WithValue(context.Background(), contextSignerKey("test"), c.bearerSuffix),
+			)
+
+			if _, err := tr.RoundTrip(req); err != nil {
+				t.Fatalf("error calling RoundTrip: %v", err)
+			}
+
+			if !check.Captured {
```

Comment:
nit: you can omit this - check.Value will be nil if this is not set, and we're always comparing against a populated want value.

Reply:
That's quite true. I added this so the test failure was more self explanatory even though it's not strictly necessary. Happy to remove it though, LMK what you think.